### PR TITLE
Updating Docs to add details on OLM use case

### PR DESCRIPTION
**Pull request by sujil02**
- Adding details about Understanding sample OLM use cases to get started. 
- Basics on when to use OLM.
- Some use cases on benefits overlap with ```Why should I use OLM``` subtitle hence eliminated to remove redundancy

### CODE REVIEW

File: content/en/docs/_index.md [hunk 1]
```diff
@@ -12,6 +12,12 @@ menu:
 
 Read more in the [introduction blog post](https://operatorhub.io/what-is-an-operator).
 
+## When to use OLM
+
+
+The premise of an Operator is to have it be a custom form of Controllers, a core concept of Kubernetes. A controller is basically a software loop that runs continuously on the Kubernetes master nodes. In these loops the control logic looks at certain Kubernetes objects of interest. It audits the desired state of these objects, expressed by the user, compares that to what’s currently going on in the cluster and then does anything in its power to reach the desired state.
+This declarative model is basically the way a user interacts with Kubernetes. Operator solves this problem and basically replaces a "human" operator responsible to perform certain task like provisioning and configuring with a "software" operator.
```
Comment (jmrodri) [inline content/en/docs/_index.md:19]: This doesn't really explain to me "when to use OLM", it simply states what an Operator is.

File: content/en/docs/_index.md [hunk 2]
```diff
@@ -29,5 +35,13 @@ Operators must claim ownership of their APIs. OLM will prevent conflicting Opera
 ### Declarative UI controls
 Operators can behave like managed service providers. Their user interface on the command line are APIs. For graphical consoles OLM annotates those APIs with descriptors that drive the creation of rich interfaces and forms for users to interact with the Operator in a natural, cloud-like way. 
 
+
+## Understanding sample OLM use-cases to get started.
+
+Operator Lifecycle Manager (OLM) helps users install, update, and manage the lifecycle of all Operators and their associated services running across their clusters.
+Operator solves this problem and basically replaces this "human" operator with a "software" operator. At its core it has the same control loop mechanism that Kubernetes has, that watches for changes in the application state.
+
+- For developers, a self-service experience allows provisioning and configuring instances of databases, monitoring without having to be subject matter experts, because the Operator has that knowledge baked into it.
```
Comment (jmrodri) [inline content/en/docs/_index.md:44]: This doesn't really give me a sample use case for using OLM.

**Review by jmrodri — CHANGES_REQUESTED** [review]
The two main paragraphs don't really go with the subtitles listed.

File: content/en/docs/_index.md (hunk 2 above)
Comment (sujil02) [reply content/en/docs/_index.md:44]: Thank you for the comment. I am working on improving the article.

File: content/en/docs/_index.md (hunk 1 above)
Comment (sujil02) [reply content/en/docs/_index.md:19]: Thank you for the comment. I am working on improving the article.

**Review by marc-obrien — CHANGES_REQUESTED** [review]

File: content/en/docs/_index.md [hunk 3]
```diff
@@ -29,5 +35,18 @@ Operators must claim ownership of their APIs. OLM will prevent conflicting Opera
 ### Declarative UI controls
 Operators can behave like managed service providers. Their user interface on the command line are APIs. For graphical consoles OLM annotates those APIs with descriptors that drive the creation of rich interfaces and forms for users to interact with the Operator in a natural, cloud-like way. 
 
+
+## Understanding sample OLM use-cases to get started.
+
+A user can expect the need of an operator in multiple senario some may be :-
```
[the remark this answers is not in the document]
Comment (sujil02) [reply content/en/docs/_index.md:41]: done

File: content/en/docs/_index.md [hunk 4]
```diff
@@ -29,5 +35,18 @@ Operators must claim ownership of their APIs. OLM will prevent conflicting Opera
...
+- operator deploying an application on demand
```
[the remark this answers is not in the document]
Comment (sujil02) [reply content/en/docs/_index.md:42]: Done

File: content/en/docs/_index.md [hunk 5]
```diff
@@ -29,5 +35,18 @@ Operators must claim ownership of their APIs. OLM will prevent conflicting Opera
...
+- taking and restoring backups of that application's state
+- handling upgrades of the application code alongside related changes such as database schemas or extra configuration settings
+- publishing a Service to applications that don't support Kubernetes APIs to discover them
+- simulating failure in all or part of your cluster to test its resilience
+
+Now in a multi-cluster enviornment there can be multiple such operators performing varios dedicated tasks, but as we scale and design more of these operators the use case of managing the operators becomes evidently important and addressing this the Operator Lifecycle Manager (OLM) helps users install, update, and manage the lifecycle of all Operators and their associated services running across their clusters hence enhancing the scalablity further.
```
[the remark this answers is not in the document]
Comment (sujil02) [reply content/en/docs/_index.md:48]: Done.

**Comment by theishshah** [comment]
can you re-base this since we've had some changes on the docs/_index in master since this was opened - if your changes are still valid/all good after that we can move to merge

**Comment by anik120** [comment]
please /hold

till we can discuss the current PRs coming in to this repo.

**Pull request merged**